pygccxml/declarations/type_traits.py

```diff
@@ -297,6 +303,16 @@ def remove_const(type):
     if not is_const(nake_type):
         return type
     else:
+        if isinstance(nake_type, cpptypes.array_t):
+            is_v = is_volatile(nake_type)
+            if is_v:
+                base_type = nake_type.base.base.base
+            else:
+                base_type = nake_type.base.base
+            result_type = base_type
+            if is_v:
+                result_type = cpptypes.volatile_t(result_type)
```

Comment:
I'm not sure I understand. If you have a type, say `t`, which is `const int`, then `is_same(t, remove_const(t))` will return `False`, which is correct. But `is_same(remove_const(t), remove_const(t))` should still return `True` (but I haven't tested this).

I don't think you can get these right without creating new instances; what you have is nesting like this - `array_t -> const_t -> (type)`, and you need to remove the `const_t` from that hierarchy. But if you can think of a cleaner way to do this, I'm all for it. I really dislike the `nake_type.base.base.base` part.